runtime/pkg/rilltime/rilltime.go

```diff
@@ -196,149 +263,362 @@ func ParseCompatibility(timeRange, offset string) error {
+		if t.IncludeCurrent {
+			end = timeutil.OffsetTime(end, t.tg, 1)
 		}
-		if e.AtModifiers.AnchorOverride.AbsTime != nil {
-			absTm, _ := time.Parse("2006-01-02 15:04", *e.AtModifiers.AnchorOverride.AbsTime)
-			return absTm, e.AtModifiers.AnchorOverride
+
+		return start, cur, end
+	}
+
+	if t.Prefix == nil {
+		// Without a prefix of either +/- we actually need a time range.
+		// EG: 7D is 7 day period.
+
+		end = timeutil.TruncateTime(cur, t.tg, tz, evalOpts.FirstDay, evalOpts.FirstMonth)
+		if t.IncludeCurrent {
+			end = timeutil.OffsetTime(end, t.tg, 1)
+			cur = timeutil.OffsetTime(cur, t.tg, 1)
 		}
+
+		start = timeutil.OffsetTime(end, t.tg, -t.offset)
+
+		return start, cur, end
 	}
 
-	if e.End == nil {
-		return evalOpts.Watermark, &TimeAnchor{
-			Watermark: true,
+	switch *t.Prefix {
+	case "-":
+		offset := t.offset
+		if t.IncludeCurrent {
+			offset--
 		}
+
+		start = timeutil.TruncateTime(cur, t.tg, tz, evalOpts.FirstDay, evalOpts.FirstMonth)
+		start = timeutil.OffsetTime(start, t.tg, -offset)
+
+		end = timeutil.OffsetTime(start, t.tg, 1)
+
+		cur = timeutil.OffsetTime(cur, t.tg, -offset)
+
+	case "+":
+		start = timeutil.TruncateTime(cur, t.tg, tz, evalOpts.FirstDay, evalOpts.FirstMonth)
+		start = timeutil.OffsetTime(start, t.tg, t.offset)
+
+		end = timeutil.OffsetTime(start, t.tg, 1)
+
+		cur = timeutil.OffsetTime(cur, t.tg, t.offset)
+
+	case "<":
+		// Anchor the range to the beginning of the higher order start
+		// EG: <4d of M : gives 1st 4 days of the current month regardless of current date.
+
+		// Anchoring to start should follow week rules https://en.wikipedia.org/wiki/ISO_week_date#First_week
+		start = truncateWithCorrection(start, t.tg, tz, evalOpts.FirstDay, evalOpts.FirstMonth)
+		end = timeutil.OffsetTime(start, t.tg, t.offset)
+		end = timeutil.TruncateTime(end, t.tg, tz, evalOpts.FirstDay, evalOpts.FirstMonth)
+
+		cur = start
+
+	case ">":
+		// Anchor the range to the end of the higher order end
+		// EG: >4d of M : gives last 4 days of the current month regardless of current date.
+		end = timeutil.CeilTime(end, t.tg, tz, evalOpts.FirstDay, evalOpts.FirstMonth)
+		start = timeutil.TruncateTime(end, t.tg, tz, evalOpts.FirstDay, evalOpts.FirstMonth)
+		start = timeutil.OffsetTime(start, t.tg, -t.offset)
+
+		cur = end
 	}
 
-	if e.End.Latest {
-		// if end has latest mentioned then start also should be relative to latest.
-		return evalOpts.MaxTime, e.End
+	return start, cur, end
+}
+
+func (t *TimeAnchor) grain() timeutil.TimeGrain {
+	// If a single unit of grain is requested, return a grain less that the part's grain
+	// But only applies to grains and not period-to-grains
+	if t.offset <= 1 && t.Grain != nil {
+		return lowerOrderMap[t.tg]
 	}
-	if e.End.Now {
-		// if end has now mentioned then start also should be relative to latest.
-		return evalOpts.Now, e.End
+	return t.tg
+}
+
+// TODO: reuse code from duration.ParseISO8601
```

Comment:
Should this be addressed now?

Reply:
I dunno how stable this syntax will be. We can do a pass to cleanup and potentially merge this with duration package.